src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/uploaddocuments/UploadDocumentFurtherEvidenceMidEventHandler.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.hmcts.reform.sscs.ccd.presubmit.uploaddocuments;
+
+import static java.util.Objects.requireNonNull;
+import static uk.gov.hmcts.reform.sscs.ccd.presubmit.uploaddocuments.DocumentType.REQUEST_FOR_HEARING_RECORDING;
+import static uk.gov.hmcts.reform.sscs.util.SscsUtil.getPartiesOnCase;
+
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.ccd.callback.Callback;
+import uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.*;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.PreSubmitCallbackHandler;
+
+@Service
+@Slf4j
+public class UploadDocumentFurtherEvidenceMidEventHandler implements PreSubmitCallbackHandler<SscsCaseData> {
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        requireNonNull(callback, "callback must not be null");
+        requireNonNull(callbackType, "callbacktype must not be null");
+
+        return (callbackType.equals(CallbackType.MID_EVENT)
+                && callback.getEvent() == EventType.UPLOAD_DOCUMENT_FURTHER_EVIDENCE);
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback,
+                                                          String userAuthorisation) {
+        if (!canHandle(callbackType, callback)) {
+            throw new IllegalStateException("Cannot handle callback.");
+        }
+
+        final CaseDetails<SscsCaseData> caseDetails = callback.getCaseDetails();
+        final SscsCaseData sscsCaseData = caseDetails.getCaseData();
+        PreSubmitCallbackResponse<SscsCaseData> callbackResponse = new PreSubmitCallbackResponse<>(sscsCaseData);
+
+        log.info(String.format("Handling uploadDocumentFurtherEvidence event for caseId %s", sscsCaseData.getCcdCaseId()));
+
+        if (showRequestHearingsPage(callbackResponse)) {
+            sscsCaseData.getSscsHearingRecordingCaseData().setShowRequestingPartyPage(YesNo.YES);
+            setPartiesToRequestInfoFrom(sscsCaseData);
+        }
+
+        return callbackResponse;
+    }
+
+    private boolean showRequestHearingsPage(PreSubmitCallbackResponse<SscsCaseData> callbackResponse) {
+
+        int requestHearingCount = 0;
+        if (callbackResponse.getData().getDraftSscsFurtherEvidenceDocument() != null) {
+            for (SscsFurtherEvidenceDoc sscsFurtherEvidenceDoc : callbackResponse.getData().getDraftSscsFurtherEvidenceDocument()) {
```

Comment:
nice to have: use the functional style to get the requestHearingCount i.e. stream, filter, sum.
nice to have: put this in a separate method that describes what it does.

Reply:
Good shout - updated